pytensor/link/jax/dispatch/pad.py

```diff
@@ -0,0 +1,20 @@
+import jax.numpy as jnp
+
+from pytensor.link.jax.dispatch import jax_funcify
+from pytensor.tensor.pad import Pad
+
+
+fixed_kwargs = {"reflect": ["reflect_type"], "symmetric": ["reflect_type"]}
+
+
+@jax_funcify.register(Pad)
+def jax_funcify_pad(op, **kwargs):
+    pad_mode = op.pad_mode
+    expected_kwargs = fixed_kwargs.get(pad_mode, {})
+    mode_kwargs = {kwarg: getattr(op, kwarg) for kwarg in expected_kwargs}
```

Comment:
Is this a bit over-engineered now that we don't have odd?

Reply:
I'd say it's "aspirationally engineered"

Reply:
serious answer: you would actually ask for `odd` if you knew you were going to compile to JAX. Might try to allow that somehow?

Reply:
You would need an invalid OpFromGraph for that, so I would suggest not.

Reply:
Ok even better answer: this is needed to forward mode-specific keyword arguments to the jax function. Trying to do it right gives an error:

```python
from pytensor.tensor.pad import Pad, allowed_kwargs


@jax_funcify.register(Pad)
def jax_funcify_pad(op, **kwargs):
    pad_mode = op.pad_mode
    expected_kwargs = allowed_kwargs.get(pad_mode, {})

    def pad(x, pad_width, *args):
        mode_kwargs = dict(zip(expected_kwargs, args))
        return jnp.pad(x, pad_width, mode=pad_mode, **mode_kwargs)

    return pad
```

```
E     ValueError: Non-hashable static arguments are not supported, as this can lead to unexpected cache-misses. Static argument (index 4) of type <class 'jaxlib.xla_extension.ArrayImpl'> for function _pad is non-hashable.
```

Does jax not allow **kwargs?

Reply:
Why do you have args in pad inner function?

Reply:
You know all the kwargs from the Op

Reply:
The kwargs change based on the mode

Reply:
I don't understand why you need to evaluate / build any keyword arguments in the inner function. Nothing depends on runtime values

Reply:
Depending on the mode, the inner function signature changes

Reply:
This is not allowed?

```python

def pad_funcify(op, node)
  pad_mode = op.pad_mode
  extra_kwargs = {}
  if pad_mode in ("reflection", "symmetric"):
    extra_kwargs["reflect_type"] = op.reflect_type
    
  
  def pad(arr, pad_width):
    return jax.pad(arr, pad_width, mode=pad_node, **extra_kwargs)

  return pad
```

Reply:
Ah if the number of arguments change you can return a different function?

```python

def funcify_pad(op, node):
  if op.pad_mode == ”x”:
    def pad(x, y, z):
      ...
  else:
    def pad (x, y)
      ..

  return pad 
```

Reply:
I think you'll agree that what I landed on in the end for this is a true work of art

Reply:
I think it's readable

Reply:
It has a certain poetic cadence